src/Rfc6455/Message.php

```diff
@@ -84,20 +86,27 @@ public function removeFromBuffer(Frame $frame) : string
      * @return Message
      * @throws \InvalidArgumentException
      * @throws LimitationException
+     * @throws WrongEncodingException
      */
     public function addFrame(Frame $frame) : Message
     {
         if ($this->isComplete) {
             throw new \InvalidArgumentException('The message is already complete.');
         }
 
-        if (count($this->frames) > 19) {
+        if (count($this->frames) > Message::MAX_MESSAGES_BUFFERING) {
```

Comment:
The exception message talks about 20 frames and your condition is 1000 frames ... You must change the message exception . I propose to make it more flexible with the value of the constant.